app/models/move.rb

```diff
@@ -152,12 +152,23 @@ def rejected?
     cancellation_reason == CANCELLATION_REASON_REJECTED
   end
 
-  def existing
-    self.class.not_cancelled.find_by(date: date, profile_id: profile_id, from_location_id: from_location_id, to_location_id: to_location_id)
+  def existing_moves
+    Move
```

Comment:
maybe this should be memoized to avoid doing duplicate queries?

Reply:
@cwrw good question; I was in two minds about this: on the one hand, memoising is more efficient. On the other hand, the memoised method won't catch a newly created duplicate that were added during the lifetime of the move instance.

I've now memoised - lets see how that works - or if unexpected duplicates slip through